pytorch_lightning/trainer/trainer.py

```diff
@@ -2001,7 +2001,9 @@ def devices(self) -> Optional[Union[List[int], str, int]]:
 
     @property
     def data_parallel_device_ids(self) -> Optional[List[int]]:
-        return self._accelerator_connector.parallel_device_ids
+        return (
+            self._accelerator_connector.parallel_device_ids if self._accelerator_connector.parallel_device_ids else None
+        )
```

Comment:
Why not this?

```suggestion
    def data_parallel_device_ids(self) -> List[int]:
        return self._accelerator_connector.parallel_device_ids
```

Reply:
We have several tests what to assert data_parallel_device_ids is None instead of []
This property `data_parallel_device_ids` could be deprecated after this PR, so I think it not worth to change the tests?

Reply:
Sounds good